### DISCUSSION TITLE
Implement number format for input fields

### DISCUSSION THREAD

**Discussion by DennisNissenUX** [Budibase/budibase#7529] [Help]
Hey guys,

I have a more specific question about the input field settings.

I am building an application in budibase where many amounts of money are specified. It would be much more user friendly if these input fields were formatted accordingly. But it is important that it is really only the display of the amount of money. I use the entered values for further calculations in the application, so it is important that the value is correct (and not e.g. as a self-formulated string) in the database.

It should work something like this. The user enters the value like this:
![Bildschirmfoto 2022-08-30 um 10 13 43](https://user-images.githubusercontent.com/105709029/187385718-12d37f57-6389-4d9e-b7cd-4c2ed3dccf07.png)

And this is how it should look when the input field is no longer selected:
![Bildschirmfoto 2022-08-30 um 10 15 09](https://user-images.githubusercontent.com/105709029/187386030-4e82d38c-f4fc-4fcb-bc8c-cd03438157b7.png)

When the user clicks into the input field again, only the pure number should be displayed again.

Does anyone know a way to display this?

If not, do you think it could be useful to develop this feature for number input fields, since the formatting of money amounts is actually a standard requirement for many applications?

As always, looking forward to your comments and ideas.

**Comment by andz-bb** [comment]
Hey @DennisNissenUX

Unfortunately I wasn't able to see a way of formatting the display of the field itself but if it's useful to you, you can use JS bindings to display the formatted value in another component. 

![currency demo](https://user-images.githubusercontent.com/110921612/187657866-6a772d6e-f060-453b-bc8e-7eb38835b3ef.gif)

Here's the code that I used to format the number as a localised currency:

`return new Intl.NumberFormat('de-DE', { style: 'currency', currency: 'EUR' }).format($("Form.Fields.Cost"));`

You may also be able to use conditional rendering to make the formatted value show up when you need it to.

The next big release of Budibase will include the ability to develop and import custom components so a form field with the ability to display a formatted value or a specific currency field could be on the way soon even if it isn't added to our core components :)

Sorry it wasn't exactly what you're looking for but hopefully it's of some help - if you have any questions let me know and I'll try and assist!
